flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/runtime/stream/sql/FunctionITCase.java

```diff
@@ -1542,6 +1542,19 @@ void testUsingAddJar() throws Exception {
                 "drop function lowerUdf");
     }
 
+    @Test
```

Comment:
Added new test case to verify that the job submission succeeds with add jar and checkpointing. Note that this test case will fail in 1.20.x and below versions.